lib/active_record/active_record_extension.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module ActiveRecordExtension
+  def self.included(base)
+    class << base
+      prepend(ClassMethods)
+    end
+  end
+
+  module ClassMethods
+    def inherited(subclass)
+      super
+
+      begin
+        # Paper trail should be enabled by default even if the class is not declared in the config.
+        # Object modification (actions :update, :touch) will not cause log saving if the class was declared with a 'false' value in the config.
+        options = {}
+        options[:class_name] = 'AuditLogItem' if defined?(AuditLogItem)
+        options[:on] = [:create, :destroy] if Rails.configuration.audit[subclass.name&.gsub('::', '/')] == false
+        subclass.send(:has_paper_trail, options) if %w[AuditLogItem PaperTrail::Version].exclude?(subclass.name)
```

Comment:
could we use paper_trail feature https://github.com/paper-trail-gem/paper_trail#per-class instead of monkey patching of activerecord? IMHO: this will be better

so leave paper_trail declaration in models as is, in a new initializer read audit.yml & disable specified models

Reply:
@gigorok PaperTrail.request is designed to be used only per request since configuration stored in RequestStore